### PR TITLE
lazy load efSearch

### PR DESCRIPTION
**Pull request by vamshin**
*Issue #, if available:*
https://github.com/opendistro-for-elasticsearch/k-NN/issues/51

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/main/java/com/amazon/opendistroforelasticsearch/knn/index/KNNIndexCache.java [hunk 1]
```diff
@@ -236,4 +236,8 @@ public void onFileDeleted(Path indexFilePath) {
             getInstance().cache.invalidate(indexFilePath.toString());
         }
     };
+
+    private String[] getQueryParams(String indexName) {
+        return new String[] {"efSearch=" + KNNSettings.getEfSearchParam(indexName)};
```
Comment (jmazanec15) [inline src/main/java/com/amazon/opendistroforelasticsearch/knn/index/KNNIndexCache.java:241]: Could you explain more why there is only 1 query parameter being passed in? Why are M and efConstruction not gotten?

File: src/main/java/com/amazon/opendistroforelasticsearch/knn/index/v1736/KNNIndex.java [hunk 2]
```diff
@@ -87,12 +87,6 @@ public long getIndexSize() {
     public void close() {
         Lock writeLock = readWriteLock.writeLock();
         writeLock.lock();
-
```
Comment (jmazanec15) [inline src/main/java/com/amazon/opendistroforelasticsearch/knn/index/v1736/KNNIndex.java:90]: Why is this no longer needed?

**Review by jmazanec15 — COMMENTED** [review]
Just have a couple questions. Overall looks good to me

File: src/main/java/com/amazon/opendistroforelasticsearch/knn/index/v1736/KNNIndex.java (hunk 2 above)
Comment (vamshin) [reply src/main/java/com/amazon/opendistroforelasticsearch/knn/index/v1736/KNNIndex.java:90]: Good catch. Bad merge. Will add it back.

File: src/main/java/com/amazon/opendistroforelasticsearch/knn/index/KNNIndexCache.java (hunk 1 above)
Comment (vamshin) [reply src/main/java/com/amazon/opendistroforelasticsearch/knn/index/KNNIndexCache.java:241]: ```efSearch``` is required only during searches.  Other algorithm params are needed for indexing.

**Review by jmazanec15 — APPROVED** [review]
LGTM thanks!

**Pull request merged**